### ISSUE TITLE
[bug] livereload uses port 0 when website is on default ports

### ISSUE THREAD

**Issue by rogergrambihler** [lepture/python-livereload#194]
**Steps**
- Start a python server using live reload on http://localhost:50000
- Use ngrok to so Url to browse to your site is https://somehost

**Results**

livereload is trying to connect on port 0 instead of the default port for the protocol -- https://somehost:0
  
**Expected Behavior**

Expected livereload to attempt to connect on https://somehost:443

**More Info**

Chrome and other browsers return an empty string for window.location.port when url uses the default port such as http://localhost or https://localhost.

Proposed fix would be to as part of the live_script check for window.location.port and if an empty string set the port to the default value for the protocol.

With this code change live reload works correctly through ngrok and other reverse proxy type configs where the Url form the browser uses the default port

```py
live_script = (
    '<script type="text/javascript">(function(){'
    'var s=document.createElement("script");'
    'var port=%s;'
    's.src="//"+window.location.hostname+":"+port'
    '+ "/livereload.js?port=" + port;'
    'document.head.appendChild(s);'
    '})();</script>'
)
if liveport:
    live_script = escape.utf8(live_script % liveport)
else:
    live_script = escape.utf8(live_script % '("" == window.location.port) ? ((window.location.protocol == "https:") ? 443: 80) : window.location.port')
```

**Comment by daqingshu** [comment]
#195 fixed

**Comment by boskkrkbuh** [comment]
In file server.py where the variable live_script is defined:
after    `'var port=%s;'`
insert  `'if (port == "") port=(window.location.protocol == "https:") ? "443" : "80";'`

This resolves the issue.

**Comment by pradyunsg** [comment]
Fixed by #195 as far as I can tell.
